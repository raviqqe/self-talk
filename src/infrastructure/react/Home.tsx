import React from "react";
import styled from "styled-components";
import { CreateDocument } from "./CreateDocument";
import { Documents, IProps as IDocumentsProps } from "./Documents";
import { SignOut } from "./SignOut";
import { darkYellow } from "./style/colors";

const Container = styled.div`
  display: flex;
  flex-direction: column;
  height: 100vh;
  width: 85ex;
  max-width: 100%;
  margin: auto;

  > :first-child {
    flex: 1;
  }
`;

const CreateDocumentContainer = styled.div`
  position: relative;

  * {
    z-index: 1;
  }
`;

const CreateDocumentBackground = styled.div`
  background: ${darkYellow};
  position: absolute;
  top: 0;
  left: -100vw;
  width: 200vw;
  height: 100%;
  box-shadow: 0rem 0rem 0.5rem rgba(0, 0, 0, 0.3);
  z-index: 0;
`;

const SignOutContainer = styled.div`
  position: fixed;
  top: 0.5em;
  right: 0.5em;
`;

export interface IProps extends IDocumentsProps {
  createDocument: (text: string) => Promise<void>;
  signOut: () => void;
}

export const Home = ({
  createDocument,
  documents,
  insertFiles,
  listDocuments,
  listMoreDocuments,
  signOut,
  updateDocument,
}: IProps) => (
  <Container>
    <Documents
      documents={documents}
      insertFiles={insertFiles}
      listDocuments={listDocuments}
      listMoreDocuments={listMoreDocuments}
      updateDocument={updateDocument}
    />
    <CreateDocumentContainer>
      <CreateDocument
        createDocument={createDocument}
        insertFiles={insertFiles}
      />
      <CreateDocumentBackground />
    </CreateDocumentContainer>
    <SignOutContainer>
      <SignOut signOut={signOut} />
    </SignOutContainer>
  </Container>
);
