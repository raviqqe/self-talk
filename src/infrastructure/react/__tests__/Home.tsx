import {
  cleanup,
  fireEvent,
  render,
  waitForDomChange
} from "@testing-library/react";
import React from "react";
import { create } from "react-test-renderer";
import { Home } from "../Home";

afterEach(cleanup);

it("renders", () => {
  expect(
    create(
      <Home
        createDocument={async () => null}
        insertImages={async () => "url"}
        listDocuments={async function*() {
          yield [];
        }}
        signOut={async () => undefined}
        updateDocument={async () => null}
      />
    ).toJSON()
  ).toMatchSnapshot();
});

it("creates a document", async () => {
  const createDocument = jest.fn(async () => ({
    createdAt: 42,
    id: "id",
    text: "text"
  }));

  const { container, getByLabelText } = render(
    <Home
      createDocument={createDocument}
      insertImages={async () => "url"}
      listDocuments={async function*() {
        yield [];
      }}
      signOut={async () => undefined}
      updateDocument={async () => null}
    />
  );

  await waitForDomChange({ container });

  fireEvent.change(container.querySelector("textarea") as HTMLTextAreaElement, {
    target: { value: "foo" }
  });

  fireEvent.click(getByLabelText("Create"));

  expect(createDocument.mock.calls).toHaveLength(1);
});

it("updates a document", async () => {
  const updateDocument = jest.fn(async () => ({
    createdAt: 42,
    id: "id",
    text: "text"
  }));

  const { container, getByLabelText } = render(
    <Home
      createDocument={async () => null}
      insertImages={async () => "url"}
      listDocuments={async function*() {
        yield [{ id: "id", text: "text", createdAt: 42 }];
      }}
      signOut={async () => undefined}
      updateDocument={updateDocument}
    />
  );

  await waitForDomChange({ container });

  fireEvent.click(getByLabelText("Edit"));

  fireEvent.change(container.querySelector("textarea") as HTMLTextAreaElement, {
    target: { value: "foo" }
  });

  fireEvent.click(getByLabelText("Save"));

  expect(updateDocument.mock.calls).toHaveLength(1);
});
