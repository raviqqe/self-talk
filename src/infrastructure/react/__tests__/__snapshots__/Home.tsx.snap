// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="sc-fzokOt Qcwer"
>
  <div
    className="sc-fzplWN gLelMI"
  >
    <div
      className="sc-fznKkj HfYSd"
    />
  </div>
  <div
    className="sc-fzqBZW hmZiwB"
  >
    <div
      className="sc-AxhUy jkncAu"
    >
      <textarea
        async={true}
        className="sc-AxiKw emBwCb sc-AxgMl cNIsWQ"
        onChange={[Function]}
        onDrop={[Function]}
        onKeyDown={[Function]}
        onPaste={[Function]}
        placeholder="Write in Markdown ..."
        rows={1}
        value=""
      />
      <button
        aria-label="Create"
        className="sc-AxjAm sc-AxirZ ekcjUO"
        onClick={[Function]}
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
          />
        </svg>
      </button>
    </div>
    <div
      className="sc-fzqNJr fCAtrr"
    />
  </div>
  <div
    className="sc-fzoyAV eGeOGW"
  >
    <button
      className="sc-AxjAm sc-AxirZ huBbKx"
      onClick={[Function]}
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
        />
      </svg>
    </button>
  </div>
</div>
`;
